packages/slice-machine/src/modules/selectedSlice/sagas.ts

```diff
@@ -0,0 +1,328 @@
+    if (response.status > 209) {
+      return setData({
+        loading: false,
+        done: true,
+        error: response.data.err,
+        status: response.status,
+        message: response.data.reason,
+        imageLoading: false,
+      });
+    }
+    setData({
+      loading: false,
+      done: true,
+      error: null,
+      warning: !!response.data.warning,
+      status: response.status,
+      message: response.data.warning || "Screenshots were saved to FileSystem",
+      imageLoading: false,
+    });
+    yield put(
+      generateSliceScreenshotCreator.success({
+        screenshots: response.data.screenshots,
+        component,
+      })
+    );
+  } catch (e) {
+    yield put(
+      openToasterCreator({
+        message: "Internal Error: Screenshot not saved",
+        type: ToasterType.ERROR,
+      })
+    );
+  }
+}
+
+export function* generateSliceCustomScreenshotSaga({
+  payload,
+}: ReturnType<typeof generateSliceCustomScreenshotCreator.request>) {
+  const { variationId, component, setData, file } = payload;
+  try {
+    const form = new FormData();
+    form.append("file", file);
+    form.append("libraryName", component.from);
+    form.append("sliceName", component.model.name);
+    form.append("variationId", variationId);
+    setData({
+      loading: true,
+      done: false,
+      error: null,
+      status: null,
+      imageLoading: true,
+    });
+    const response = (yield call(
+      generateSliceCustomScreenshotAxios,
+      form
+    )) as SagaReturnType<typeof generateSliceCustomScreenshotAxios>;
+    if (response.status > 209) {
+      return setData({
+        loading: false,
+        done: true,
+        status: response.status,
+        imageLoading: false,
+      });
+    }
+    setData({
+      loading: false,
+      done: true,
+      error: null,
+      status: response.status,
+      message: "New screenshot added!",
+      imageLoading: false,
+    });
+    yield put(
+      generateSliceCustomScreenshotCreator.success({
+        variationId,
+        screenshot: response.data,
+        component,
+      })
+    );
+  } catch (e) {
+    yield put(
+      openToasterCreator({
+        message: "Internal Error: Custom screenshot not saved",
+        type: ToasterType.ERROR,
+      })
+    );
+  }
+}
+
+export function* saveSliceSaga({
+  payload,
+}: ReturnType<typeof saveSliceCreator.request>) {
+  const { extendedComponent, setData } = payload;
+  try {
+    setData({
+      loading: true,
+      done: false,
+      error: null,
+      status: null,
+      message: null,
+    });
+    const response = (yield call(
+      saveSliceAxios,
+      extendedComponent
+    )) as SagaReturnType<typeof saveSliceAxios>;
+    if (response.status > 209) {
+      return setData({
+        loading: false,
+        done: true,
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
```

Comment:
any reason you have eslint disabled here but not in the other sagas?

Reply:
the return type of this one is `Promise<AxiosResponse>` whereas other ones have more specific types, e.g. `Promise<AxiosResponse<ScreenshotUI>>`